src/player_entry.py

```diff
@@ -66,16 +67,14 @@ def on_tab(event: tk.Event, root: tk.Tk, supabase_client, entry_ids: Dict, users
             username: str = database_response.data[0]["username"]
             
             # If user has already been entered, display error message and refocus entry field to clear input
-            for team in users:
-                for eq_id in users[team]:
-                    if user_id == users[team][eq_id][0]:
-                        messagebox.showerror("Error", "User has already been entered")
-                        event.widget.delete(0, tk.END)
-                        root.after_idle(lambda: event.widget.focus_set())
-                        return
+            if user_id in [user.user_id for user in users["green"]] or user_id in [user.user_id for user in users["red"]]:
+                messagebox.showerror("Error", "User ID has already been entered")
+                event.widget.delete(0, tk.END)
+                root.after_idle(lambda: event.widget.focus_set())
+                return
```

Comment:
Attempted to add some getters and setters to the user class so the logic for bounds checking on user input could be moved there, but it would require the users dictionary to be passed in to ensure there were no repeated inputs. Decided to keep all of it here in the on_tab function instead for consistency.